internal/retriever/smart_mixed.go

```
@@ -0,0 +1,59 @@
+package retriever
```

Ignoring errors returned from another function call can hide issues. At minimum log the error.